url9d04094941_petermikitsh_PizzaDeliverySystem_tgz-pJ8-testdrivers_ManagerReportTestDriverJ8/src/ninja/OvenRunner.java

```diff
@@ -0,0 +1,125 @@
+package ninja;
+
+import model.FoodItem;
+import model.Oven;
+
+/**
+ * An oven runner is responsible for transferring Food items from the awaiting
+ * oven space to the oven
+ * 
+ * @author  Isioma Nnodum	iun4534@rit.edu
+ */
+public class OvenRunner extends Thread {
+
+	public boolean _running;
+
+	public FoodItem _currentFoodItem;
+
+	/**
+	 * Default constructor.
+	 */
+	public OvenRunner() {
+		_running = false;
+	}
+
+	/**
+	 * Deactivate this runner.
+	 */
+	public void deactivate() {
+		_running = false;
+	}
+
+	/**
+	 * Get the FoodItem this runner is running.
+	 * 
+	 * @return
+	 */
+	public FoodItem getCurrentFoodItem() {
+		return _currentFoodItem;
+	}
+
+	/**
+	 * Forcibly modify the FoodItem this running is running.
+	 */
+	public void setCurrentFoodItem(FoodItem newFI) {
+		_currentFoodItem = newFI;
+	}
+
+	/**
+	 * Get the next item to be placed in oven.
+	 */
+	private void retreive() {
+
+		while ((null == _currentFoodItem) && _running) {
+			
+			_currentFoodItem = Kitchen.ovenRunnerRetrieve();
+
+			// Sleep for a bit.
+			try {
+				Thread.sleep(Kitchen.KITCHEN_THREAD_WAIT_TIME);
+			} catch (InterruptedException ex) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)